### PR TITLE
Updated openblas version to 0.3.24 in the common packages.yaml spec.

### PR DESCRIPTION
**Pull request by srherbener**
### Summary

This PR bumps the common version of openblas from 0.3.19 to 0.3.24. Hopefully, this will be acceptable across the board (simpler to maintain), but it's really only required for `apple-clang@15.x` as far as I know. I am completely open to keeping the common version on 0.3.19 and only using 0.3.24 for the Mac if that is what needs to be done.

### Testing

Tested building on my Mac using `apple-clang@15.0.0` and openblas builds successfully.

### Applications affected

List all known applications (UFS WM, JEDI, SRW, etc.) intentionally or unintentionally affected by this PR.

it appears that openblas is a dependency for a fair number of applications. Here is what is reported on my Mac when building in the unified-dev environment (spack find openblas output):
```
==> In environment /Users/steveherbener/spack-stack/envs/unified.mymacos
==> Root specs
-- no arch / apple-clang ----------------------------------------
crtm@v2.4.1-jedi%apple-clang      jedi-mpas-env%apple-clang 
crtm@2.4.0%apple-clang            jedi-neptune-env%apple-clang 
ewok-env%apple-clang              jedi-tools-env%apple-clang 
fms@release-jcsda%apple-clang     jedi-ufs-env%apple-clang 
fms@2023.01%apple-clang           jedi-um-env%apple-clang 
fms@2023.02%apple-clang           madis@4.5%apple-clang 
global-workflow-env%apple-clang   soca-env%apple-clang 
gsi-env%apple-clang               ufs-srw-app-env%apple-clang 
jedi-fv3-env%apple-clang          ufs-weather-model-env%apple-clang 

==> Installed packages
-- darwin-ventura-m1 / apple-clang@15.0.0 -----------------------
openblas@0.3.24
==> 1 installed package
```

### Systems affected

Mac and, if acceptable to bump the common packages.yaml version to 0.3.24, all the HPC platforms.

### Dependencies

- [x] waiting on https://github.com/JCSDA/spack/pull/335

### Issue(s) addressed

Fixes the build of openblas on MacOS using `apple-clang@15.0.0` compiler.

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [ ] These changes have been tested on the affected systems and applications
    Only on my Mac so far
- [ ] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

**Comment by srherbener** [comment]
For the CI test that failed, here is the error message:
```
>> 1015    /tmp/tmp-mount-MGMMwD/diskspace_workaround_spack_stack/spack-cache
             /build_stage/spack-stage-ecflow-5.8.4-3pmfndgkeqw674p7zpbvnlzep7b2
             psio/spack-build-3pmfndg/Viewer/libViewer/src/moc_LocalSocketServe
             r.cpp:10:10: fatal error: cannot open file '/tmp/tmp-mount-MGMMwD/
             diskspace_workaround_spack_stack/spack-cache/build_stage/spack-sta
             ge-ecflow-5.8.4-3pmfndgkeqw674p7zpbvnlzep7b2psio/spack-build-3pmfn
             dg/Viewer/libViewer/src/../../../../../../../../../../Users/ec2-us
             er/spack-stack/spack-cache/build_stage/spack-stage-ecflow-5.8.4-3p
             mfndgkeqw674p7zpbvnlzep7b2psio/spack-src/Viewer/libViewer/src/Loca
             lSocketServer.hpp': Permission denied
     1016    #include "../../../../../../../../../../Users/ec2-user/spack-stack
             /spack-cache/build_stage/spack-stage-ecflow-5.8.4-3pmfndgkeqw674p7
             zpbvnlzep7b2psio/spack-src/Viewer/libViewer/src/LocalSocketServer.
             hpp"
     1017             ^
     1018    1 error generated.
  >> 1019    make[2]: *** [Viewer/libViewer/src/CMakeFiles/viewer.dir/moc_Local
             SocketServer.cpp.o] Error 1
     1020    make[2]: *** Waiting for unfinished jobs....
```

Looks like the build of ecflow failed. Any ideas what to do to fix this? Or should I simply re-run the tests? Thanks!

**Comment by climbfuji** [comment]
@srherbener can you please pull in develop (in spack and spack-stack) to resolve the submodule pointer conflicts? Hopefully this was a one-off glitch. The CI tests passed on that system for the container PR that we just merged.

**Comment by climbfuji** [comment]
We should try to run the container tests as well, but we can only do that after the spack PR was merged.

**Comment by climbfuji** [comment]
CI tests passed, I merged the spack PR - please update this PR. Thanks!

File: .github/workflows/macos-ci-aarch64.yaml
Comment (climbfuji) [inline .github/workflows/macos-ci-aarch64.yaml:115]: I've added this to fix the CI problems on macOS arm64

**Review by climbfuji — APPROVED** [review]
Approved (I made the submodule pointer / .gitmodules update), waiting for CI.

**Pull request merged**
